src/Microsoft.TemplateEngine.Core/Util/ProcessorState.cs

```diff
@@ -342,9 +347,12 @@ public bool Run()
                 // Console.WriteLine("LastBlock");
                 // string text = System.Text.Encoding.UTF8.GetString(CurrentBuffer, CurrentBufferLength - toWrite, toWrite).Replace("\0", "\\0");
                 // Console.WriteLine(text);
-                _target.Write(CurrentBuffer, CurrentBufferLength - toWrite, toWrite);
+                _temporaryTarget.Write(CurrentBuffer, CurrentBufferLength - toWrite, toWrite);
             }
 
+            _temporaryTarget.Flush();
+            _temporaryTarget.Position = 0;
+            _temporaryTarget.CopyTo(_target);
```

Comment:
Should we seek to the beginning of `_target` to be safe?
```suggestion
            _target.Position = 0;
            _temporaryTarget.CopyTo(_target);
```